mypy/nodes.py

```diff
@@ -1946,6 +1946,27 @@ def __init__(self, names: 'SymbolTable', defn: ClassDef, module_name: str) -> No
             for vd in defn.type_vars:
                 self.type_vars.append(vd.name)
 
+    declared_metaclass = None  # type: Optional[mypy.types.Instance]
+    metaclass_type = None  # type: Optional[mypy.types.Instance]
+
+    def calculate_metaclass_type(self) -> 'Optional[mypy.types.Instance]':
+        if self.mro is None:
+            # XXX why does this happen?
```

Comment:
Do you have a sample that triggers this condition?

Presumably the clue is the comment in `__init__` on line 1940 above. Maybe it has something to do with the MRO being calculated twice -- the first time (in semanal.py on line 850) it may fail if there's an import cycle causing a base class being a forward reference; the second time on line 3128 in ThirdPass, called from build.py line 1440. (There's also a call from fixup.py, but I think that's not what you're seeing.)